lib/parse.js

```diff
@@ -123,16 +125,24 @@ module.exports = {
   /** Given an array of files returns the root OAS file if present
    *
    * @param {Array} filesPathArray Array of file paths
+   * @param {Object} files Files map
    * @return {String} rootFile
    */
-  getRootFiles: function (filesPathArray) {
+  getRootFiles: function (filesPathArray, files = {}) {
     let rootFilesArray = [];
-
     filesPathArray.forEach((filePath) => {
+      let obj,
+        file,
+        oasObject;
       try {
-        let file = fs.readFileSync(filePath.fileName, 'utf8'),
-          oasObject,
-          obj = this.getOasObject(file);
+        if (!_.isEmpty(files)) {
```

Comment:
@umeshp7 Clarifying comments that explain when you expect each branch here to be used

Reply:
Done